abi.md

```diff
@@ -0,0 +1,147 @@
+---
+author:
+- The FIRRTL Specification Contributors
+title: Specification for the FIRRTL ABI
+revisionHistoryAbi: true
+---
+
+# Introduction
+
+FIRRTL defines a language/IR for describing synchronous hardware circuits.  This
+document specifies the mapping of FIRRTL constructs to Verilog in a manner
+similar to an application binary interface (ABI) which enables predictability of
+the output of key constructs necessary for the interoperability between circuits
+described in FIRRTL and between other languages and FIRRTL output.
+
+This document describes multiple versions of the ABI, specifically calling
+specific changes in later versions.  It is expected that a conforming FIRRTL
+compiler can lower to all specified ABIs.  This mechanism exists to allow
+improved representations when using tools which have better Verilog support and
+allow migration of existing development flows to significant representational
```

Comment:
"allow migration.. to significant representational changes"? What does this mean?